src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1105,6 +1107,89 @@ CreateSchemaDDLCommand(Oid schemaId)
 	return schemaNameDef->data;
 }
 
+/*
+ * GrantOnSchemaDDLCommands creates a list of ddl command for replicating the permissions
+ * of roles on schemas.
+ */
+List *
+GrantOnSchemaDDLCommands(Oid schemaOid)
+{
+	HeapTuple schemaTuple = SearchSysCache1(NAMESPACEOID, ObjectIdGetDatum(schemaOid));
+	bool isNull = true;
+	Datum aclDatum = SysCacheGetAttr(NAMESPACEOID, schemaTuple, Anum_pg_namespace_nspacl, &isNull);
+	if (isNull)
+	{
+		ReleaseSysCache(schemaTuple);
+		return NIL;
+	}
+	Acl *acl = DatumGetAclPCopy(aclDatum);
+	Oid *members = NULL;
+	int memberNumber = aclmembers(acl, &members);
+	List *commands = NIL;
+	AclItem *aclDat = ACL_DAT(acl);
+	int aclNum = ACL_NUM(acl);
+	char * schemaName = get_namespace_name(schemaOid);
+
+	for (int i=0; i < memberNumber; i++)
+	{
+		List * queries = GenerateGrantOnSchemaQueryFromACL(members[i], schemaName, aclNum, aclDat);
+		commands = list_concat(commands, queries);
+	}
+	commands = list_concat(commands, GenerateGrantOnSchemaQueryFromACL(0, schemaName, aclNum, aclDat));
+
+	ReleaseSysCache(schemaTuple);
+	return commands;
+}
+
+/*
+ * GenerateGrantOnSchemaQueryFromACL generates a query string for replicating a users permissions
+ * on a schema.
+ */
+List *
+GenerateGrantOnSchemaQueryFromACL(Oid memberOid, char * schemaName, int aclNum, AclItem *aclDat)
```

Comment:
This function contains a lot of the syntax generation that overlaps with the deparser.
Would it be possible to create a `GrantStmt` data structure from the catalog structures that we could feed to the deparser to reuse that code?